### PR TITLE
HPCC-25972 Fix initial set of dependencies on queryEnvironmentConf()

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith I suspect there are more, but this is an initial set of changes to look at.  I suspect you will need better fixes when you integrate dafilesrv....

**Comment by ghalliday** [comment]
@richardkchapman you may prefer a smaller fix which removes the errors, and then have this as a separate PR that addresses some of the reasons it was failing.

**Comment by ghalliday** [comment]
How could this regression have been avoided?
- Checking that container version had been built and run
- Me being clearer on what stage the PR was at.

File: dali/server/daserver.cpp [hunk 1]
```diff
@@ -665,6 +665,7 @@ int main(int argc, const char* argv[])
 // Audit logging
         StringBuffer auditDir;
         {
+            //MORE: Does this need to change in CONTAINERIZED mode?
```
Comment (jakesmith) [inline dali/server/daserver.cpp:668]: I think it should just output to standard logging (stderr).
The audience field should be enough to discriminate it when filtering/viewing in ES etc.

File: fs/dafsclient/rmtclient.cpp [hunk 2]
```diff
@@ -91,11 +91,15 @@ static unsigned dafsConnectFailRetryTimeMs = defaultDafsConnectFailRetrySeconds
 
 MODULE_INIT(INIT_PRIORITY_DAFSCLIENT)
 {
+#ifdef _CONTAINERIZED
+    //MORE: This function is called too soon to read them from the configuration file.
```
Comment (jakesmith) [inline fs/dafsclient/rmtclient.cpp:95]: there is no clear use-case at the moment for a dafilesrv client inside HPCC (only outside from a different client, e.g. Java/Spark).
So at the moment the whole of rmtclient.cpp/rmtfile.cpp - could be compiled out if we wanted I think.
But, we may want to go via dafilesrv client connection in future for some reason, e.g. routing through for security reasons

File: roxie/ccd/ccdmain.cpp [hunk 3]
```diff
@@ -771,12 +771,14 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
...
 
         Linked<IPropertyTree> directoryTree = topology->queryPropTree("Directories");
+#ifndef _CONTAINERIZED
         if (!directoryTree)
         {
             Owned<IPropertyTree> envFile = getHPCCEnvironment();
             if (envFile)
                 directoryTree.set(envFile->queryPropTree("Software/Directories"));
         }
+#endif
```
Comment (jakesmith) [inline roxie/ccd/ccdmain.cpp:781]: Don't think it's an option to have "Directories" in roxie config in k8s/helm, so this #ifndef _CONTAINERIZED block may be better to around lines 773 to 795

File: system/jlib/jutil.cpp [hunk 4]
```diff
@@ -2453,7 +2453,10 @@ static CriticalSection envConfCrit;
 jlib_decl const IProperties &queryEnvironmentConf()
 {
 #if defined(_CONTAINERIZED) && defined(_DEBUG)
-    throwUnexpectedX("queryEnvironmentConf() callled from container system");
+    //The following line is currently hit by too many examples.  Re-enable the exception when more
+    //work has been done removing calls to getConfigurationDirectory() and other related functions.
+    //throwUnexpectedX("queryEnvironmentConf() callled from container system");
+    IERRLOG("queryEnvironmentConf() callled from container system");
```
Comment (jakesmith) [inline system/jlib/jutil.cpp:2459]: trivial: pre-existing typo 'callled'

**Review by jakesmith — APPROVED** [review]
@ghalliday - all looks sensible, minor comments only.

**Pull request merged**